contracts/exchangeIssuance/FlashMintNAV.sol

```diff
@@ -0,0 +1,407 @@
+        DEXAdapterV2.Addresses memory _dexAddresses
+    )
+        public
+    {
+        setController = _setController;
+        navIssuanceModule = _navIssuanceModule;
+        dexAdapter = _dexAddresses;
+        WETH = _dexAddresses.weth;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Withdraw tokens to selected address if they end up in the contract
+     *
+     * @param _tokens    Addresses of tokens to withdraw, specifiy ETH_ADDRESS to withdraw ETH
+     * @param _to        Address to send the tokens to
+     */
+    function withdrawTokens(IERC20[] calldata _tokens, address payable _to) external onlyOwner payable {
+        for(uint256 i = 0; i < _tokens.length; i++) {
+            if(address(_tokens[i]) == ETH_ADDRESS){
+                _to.sendValue(address(this).balance);
+            }
+            else{
+                _tokens[i].safeTransfer(_to, _tokens[i].balanceOf(address(this)));
+            }
+        }
+    }
+
+    receive() external payable {
+        // required for weth.withdraw() to work properly
+        require(msg.sender == WETH, "FlashMint: DIRECT DEPOSITS NOT ALLOWED");
+    }
+
+    /**
+     * Runs all the necessary approval functions required before issuing or redeeming 
+     * a SetToken through the NAV Issuance Module. This function needs to be called
+     * before this smart contract is used with any particular SetToken, and again
+     * whenever a new reserve asset is added.
+     *
+     * @param _setToken          Address of the SetToken being initialized
+     */
+    function approveSetToken(ISetToken _setToken) external {
+        address[] memory reserveAssets = navIssuanceModule.getReserveAssets(address(_setToken));
+        for (uint256 i = 0; i < reserveAssets.length; i++) {
+            _safeApprove(IERC20(reserveAssets[i]), address(navIssuanceModule), type(uint256).max);
+        }
+        _safeApprove(IERC20(_setToken), address(navIssuanceModule), type(uint256).max);
+    }
+
+    /**
+     * Gets the amount of Set Token expected to be issued given an exact quantity of input token.
+     * This function is not marked view, but should be static called from frontends.
+     * This constraint is due to the need to interact with the Uniswap V3 quoter contract
+    *
+    * @param _setToken              Address of the Set Token to be issued
+    * @param _inputToken            Address of the token used to pay for issuance. Use WETH address if input token is ETH.
+    * @param _inputTokenAmount      Exact amount of input token to spend
+    * @param _reserveAssetSwapData  Swap data to trade input token for reserve asset. Use empty swap data if input token is the reserve asset.
+    * 
+    * @return                       Amount of SetTokens expected to be issued
+    */
+    function getIssueAmount(
+        ISetToken _setToken,
+        address _inputToken,
+        uint256 _inputTokenAmount,
+        DEXAdapterV2.SwapData memory _reserveAssetSwapData
```

Comment:
@edkim reserve asset would be e.g. USDC? and then the regular swap data format e.g. if inputToken DAI for DAI-USDC for the given `inputTokenAmount`?

Reply:
Yes, reserve asset is USDC for USDCY. So for issuing with DAI the swapData would look something like
`{
            exchange: Exchange.UniV3,
            fees: [100],
            path: [addresses.tokens.dai, addresses.tokens.USDC],
            pool: ADDRESS_ZERO,
};`

See example in test https://github.com/IndexCoop/index-coop-smart-contracts/blob/e21b456c9a436e5d073ea4ea8ac54bc9101accb8/test/integration/ethereum/flashMintNAV.spec.ts#L336